string to_attach_name
string bin_name
